### PR TITLE
Swiftlint compliance

### PR DESCRIPTION
**Pull request by remcomokveld**
There were multiple places in the code where the generated code wasn't formatted according to the swiftlint spec. I first adjusted the tests and then added the possibility to specify wether a literal should have a space prepended when emitted to the CodeWriter. 

Thanks for your time!

### CODE REVIEW

File: Sources/CodeGeneration/CodeWriter.swift [hunk 1]
```diff
@@ -292,12 +292,12 @@ extension CodeWriter {
         return self
     }
 
-    fileprivate func emit(literal value: Any?, first: Bool = false) {
+    fileprivate func emit(literal value: Any?, first: Bool = false, trimStart: Bool = false) {
```
Comment (kyle-dorman) [inline Sources/CodeGeneration/CodeWriter.swift]: the `first` param was sort of a hacky way to trim the start. If this is a more formalized way to control if a space is added or not, then I don't think the `first` param should exist anymore.

**Comment by remcomokveld** [comment]
Yeah I was in doubt about whether that one should be removed or not. I am not if this is the more formalized way to control the space here. But at least it is more dynamic.

**Comment by kyle-dorman** [comment]
Well I am fine with adding the `trimString` control flag (although I think there is probably a more elegant solution out there) but then `first` should go away and `trimString` should be used instead. If you could add/change that, that would be awesome.

**Comment by kyle-dorman** [comment]
👍

**Pull request merged**